test/foundry/utils/version-control/ConditionalVersionControl.t.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Vm, Test } from "forge-std/Test.sol";
+import { console } from "forge-std/console.sol";
+import { TransparentUpgradeableProxyV2, ERC1967Upgrade } from "@ronin/contracts/extensions/TransparentUpgradeableProxyV2.sol";
+import { ILogic, MockLogicV1, MockLogicV2 } from "@ronin/contracts/mocks/utils/version-control/MockLogic.sol";
+import { MockActor } from "@ronin/contracts/mocks/utils/version-control/MockActor.sol";
+import { MockConditionalVersionControl, ConditionalVersionControl } from "@ronin/contracts/mocks/utils/version-control/MockConditionalVersionControl.sol";
+
+contract ConditionalVersionControlTest is Test {
+  /**
+   * @dev Error thrown when a duplicated element is detected in an array.
+   * @param msgSig The function signature that invoke the error.
+   */
+  error ErrDuplicated(bytes4 msgSig);
+  /// @dev Error of set to non-contract.
+  error ErrZeroCodeContract(address addr);
+  /// @dev Error when contract which delegate to this contract is not compatible with ERC1967
+  error ErrDelegateFromUnknownOrigin(address addr);
+  /**
+   * @dev Error indicating that a function can only be called by the contract itself.
+   * @param msgSig The function signature (bytes4) that can only be called by the contract itself.
+   */
+  error ErrOnlySelfCall(bytes4 msgSig);
+
+  address logicV1;
+  address logicV2;
```

Comment:
So this test is guaranteed for upgrading from v1 to v2, not for v2 -> v3,... etc?

Reply:
yes.